slang-checks/src/main/java/com/sonarsource/slang/checks/AllBranchesIdenticalCheck.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonarsource.slang.checks;
+
+import com.sonarsource.slang.api.IfTree;
+import com.sonarsource.slang.api.MatchCaseTree;
+import com.sonarsource.slang.api.MatchTree;
+import com.sonarsource.slang.api.Tree;
+import com.sonarsource.slang.checks.api.CheckContext;
+import com.sonarsource.slang.checks.api.InitContext;
+import com.sonarsource.slang.checks.api.SlangCheck;
+import java.util.ArrayList;
+import java.util.List;
+import org.sonar.check.Rule;
+
+import static com.sonarsource.slang.checks.utils.SyntacticEquivalence.areEquivalent;
+
+@Rule(key = "S3923")
+public class AllBranchesIdenticalCheck implements SlangCheck {
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(IfTree.class, (ctx, tree) -> {
+      if (!(ctx.parent() instanceof IfTree)) {
```

Comment:
Not sure, but what happens for nested ifs ?
`if(x)
  if(y)
    return true;
  else
    return true;
 else
  return false;
`
I think the nested one will not be considered since its parent is a `if` statement too

Reply:
Fixed in efbbd33